examples/rest-api/main.py

```diff
@@ -0,0 +1,354 @@
+    i.e, our opensource config that doesn't require any API key to run.\n
+    """
+    try:
+        if app_id is None:
+            raise HTTPException(detail="App ID not provided.", status_code=400)
+
+        if get_app(db, app_id) is not None:
+            raise HTTPException(detail=f"App with id '{app_id}' already exists.", status_code=400)
+
+        yaml_path = "default.yaml"
+
+        save_app(db, app_id, yaml_path)
+
+        return DefaultResponse(response=f"App created successfully. App ID: {app_id}")
+    except Exception as e:
+        raise HTTPException(detail=f"Error creating app: {e}", status_code=400)
+
+
+@app.post("/create/yaml", tags=["Apps"], response_model=DefaultResponse)
+async def create_app_using_custom_config(config: UploadFile, db: Session = Depends(get_db)):
+    """
+    Create a new app using YAML config.
+    """
+    try:
+        contents = await config.read()
+
+        try:
+            yaml_conf = yaml.safe_load(contents)
+            app_id = yaml_conf.get("app", {}).get("config", {}).get("id", None)
+
+            if app_id is None:
+                raise HTTPException(detail="App ID not provided.", status_code=400)
+
+            if get_app(db, app_id) is not None:
+                raise HTTPException(detail=f"App with id '{app_id}' already exists.", status_code=400)
+
+            yaml_path = f"configs/{app_id}.yaml"
+            with open(yaml_path, "w") as file:
+                file.write(str(contents, "utf-8"))
+
+            save_app(db, app_id, yaml_path)
+
+        except yaml.YAMLError as exc:
+            raise HTTPException(detail=f"Error parsing YAML: {exc}", status_code=400)
+
+        return DefaultResponse(response=f"App created successfully. App ID: {app_id}")
+    except Exception as e:
+        raise HTTPException(detail=f"Error creating app: {e}", status_code=400)
+
+
+@app.get(
+    "/{app_id}/datasources",
+    tags=["Apps"],
+)
+async def get_datasources_associated_with_app_id(app_id: str, db: Session = Depends(get_db)):
+    """
+    Get all datasources for an app.\n
+    app_id: The ID of the app. Use "default" for the default app.\n
+    """
+    try:
+        if app_id is None:
+            raise HTTPException(
+                detail="App ID not provided. If you want to use the default app, use 'default' as the app_id.",
+                status_code=400,
+            )
+
+        db_app = get_app(db, app_id)
+
+        if db_app is None:
+            raise HTTPException(detail=f"App with id {app_id} does not exist, please create it first.", status_code=400)
+
+        app = App.from_config(yaml_path=db_app.config)
+
+        response = app.get_data_sources_by_app_id(app_id=app_id)
+        return {"data": response}
+    except ValueError as ve:
+        if "OPENAI_API_KEY" in str(ve) or "OPENAI_ORGANIZATION" in str(ve):
+            raise HTTPException(
+                detail="OPENAI_API_KEY or OPENAI_ORGANIZATION not set, please set them in your environment variables.",
```

Comment:
Can you please add instructions on how one can do that. Show them command like:

```
docker run -e ENV=value <image>
```

Add the correct image name.

Reply:
Thanks! I've added a proper mapping to check if there is any ValueError for a missing API key, now it will throw an error accordingly with an example docker command using that missing key.

For example if `OPENAI_API_KEY` is missing,

Error message:
```
Please set the OPENAI_API_KEY environment variable(s) when running the Docker container.\nExample: `docker run -e OPENAI_API_KEY=xxx embedchain/app:rest-api-latest
```